cmd/helmvm/token.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+
+	"github.com/replicatedhq/helmvm/pkg/defaults"
+)
+
+var tokenCommands = &cli.Command{
+	Name:        "token",
+	Usage:       "Manage node join tokens",
+	Subcommands: []*cli.Command{tokenCreateCommand},
+}
+
+var tokenCreateCommand = &cli.Command{
+	Name:  "create",
+	Usage: "Creates a new node join token",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "role",
+			Usage: "The role of the token (can be controller or worker)",
+			Value: "worker",
+		},
+		&cli.DurationFlag{
+			Name:  "expiry",
+			Usage: "For how long the token should be valid",
+			Value: 24 * time.Hour,
+		},
+	},
+	Action: func(c *cli.Context) error {
+		if runtime.GOOS != "linux" {
+			return fmt.Errorf("token create is only supported on linux")
+		}
+		if os.Getuid() != 0 {
+			return fmt.Errorf("token create must be run as root")
+		}
+		role := c.String("role")
+		if role != "worker" && role != "controller" {
+			return fmt.Errorf("invalid role %q", role)
+		}
+		cfgpath := defaults.PathToConfig("k0sctl.yaml")
+		if _, err := os.Stat(cfgpath); err != nil {
+			if os.IsNotExist(err) {
+				logrus.Errorf("Unable to read the current cluster configuration.")
+				logrus.Errorf("Tried to read cluster config from %s", cfgpath)
+				logrus.Errorf("If you are running this command with sudo you may need")
+				logrus.Errorf("to run it as 'sudo -E' to preserve your user environment.")
+				return fmt.Errorf("k0s is not installed")
+			}
+			return fmt.Errorf("unable to stat k0sctl.yaml: %w", err)
+		}
+		logrus.Infof("Creating node join token for role %s", role)
+		if !defaults.DecentralizedInstall() {
+			logrus.Warn("You are opting out of the centralized cluster management.")
+			logrus.Warn("Through the centralized management you can manage all your")
+			logrus.Warn("cluster nodes from a single location. If you decide to move")
+			logrus.Warn("on the centralized management won't be available anymore")
+			question := &survey.Confirm{
+				Message: "Do you want to use continue ?", Default: true,
+			}
+			var moveOn bool
+			if err := survey.AskOne(question, &moveOn); err != nil {
+				return fmt.Errorf("unable to ask for user confirmation: %w", err)
+			} else if !moveOn {
+				return nil
+			}
+		}
+		dur := c.Duration("expiry").String()
+		buf := bytes.NewBuffer(nil)
+		cmd := exec.Command("k0s", "token", "create", "--expiry", dur, "--role", role)
+		cmd.Stdout = buf
+		cmd.Stderr = io.Discard
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to create token: %w", err)
+		}
+		if !defaults.DecentralizedInstall() {
+			if err := defaults.SetInstallAsDecentralized(); err != nil {
+				return fmt.Errorf("failed to set decentralized install: %w", err)
+			}
+		}
+		logrus.Infof("Token created successfully.")
```

Comment:
it doesnt seem appropriate to use logging for output in this file. how about just print to stdout?

Reply:
I have been using this across all the files, unfortunately. I can go ahead and remove it from all of them. Let me know what your thoughts are on how should we signalise things like `Warnf()` and `Errorf()` before I act on this.

Reply:
don't act on it yet but i think we should give it some more thought and maybe explore how other clis message the user. in my opinion, output that is informational for the user without verbosity increased should probably not be in log format with level and timestamp. this feedback is not for the prototype phase.